src/Components/Components/src/Rendering/Renderer.cs

```diff
@@ -150,6 +150,10 @@ protected async Task RenderRootComponentAsync(int componentId, ParameterCollecti
             try
             {
                 await ProcessAsynchronousWork();
+                if (_pendingTasks.Count != 0)
+                {
+                    throw new Exception();
+                }
```

Comment:
Is this new code block meant to be here? I'm guessing we wouldn't really want to throw a completely blank `Exception`. Doesn't the `Debug.Assert` on L157 handle this adequately already?

Reply:
Debugging something and forgot to remove it. `Debug.Asserts` just result in xunit crashing the test process and I had a bug here at some point